nupic/regions/TPRegion.py

```diff
@@ -506,6 +517,18 @@ def _compute(self, inputs, outputs):
       size = activeLearnCells.shape[0] * activeLearnCells.shape[1]
       outputs['lrnActiveStateT'][:] = activeLearnCells.reshape(size)
 
+    if self.computePredictedActiveCellIndices:
+      # Reshape so we are dealing with 1D arrays
+      activeState = self._tfdr.getActiveState().reshape(-1).astype('float32')
+      predictedState = self._tfdr.getPredictedState().reshape(-1).astype('float32')
+      activeIndices = numpy.where(activeState != 0)[0]
+      predictedIndices= numpy.where(predictedState != 0)[0]
+      predictedActiveIndices = numpy.intersect1d(activeIndices, predictedIndices)
+      outputs["activeCells"].fill(0)
+      outputs["activeCells"][activeIndices] = 1
+      outputs["predictedActiveCells"].fill(0)
+      outputs["predictedActiveCells"][predictedActiveIndices] = 1
```

Comment:
Should be "successfullyPredictedCells" to make it absolutely distinct from:

1. Predicted Cells - Cells currently predicted in the current input cycle.
2. Previously Predicted Cells - Cells predicted in the previous input cycle
(t -1)
3. Successfully Predicted Cells - Cells predicted in the previous input
cycle that **also** appear in the active input field. (as distinct from
previously predicted cells which are not successful (a.k.a. not present as
activations in current input field)

On Mon, Jun 29, 2015 at 11:18 AM, Akhila <notifications@github.com> wrote:

> In nupic/regions/TPRegion.py
> <https://github.com/numenta/nupic/pull/2206#discussion_r33481503>:
>
> > @@ -506,6 +517,18 @@ def _compute(self, inputs, outputs):
> >        size = activeLearnCells.shape[0] * activeLearnCells.shape[1]
> >        outputs['lrnActiveStateT'][:] = activeLearnCells.reshape(size)
> >
> > +    if self.computePredictedActiveCellIndices:
> > +      # Reshape so we are dealing with 1D arrays
> > +      activeState = self._tfdr.getActiveState().reshape(-1).astype('float32')
> > +      predictedState = self._tfdr.getPredictedState().reshape(-1).astype('float32')
> > +      activeIndices = numpy.where(activeState != 0)[0]
> > +      predictedIndices= numpy.where(predictedState != 0)[0]
> > +      predictedActiveIndices = numpy.intersect1d(activeIndices, predictedIndices)
> > +      outputs["activeCells"].fill(0)
> > +      outputs["activeCells"][activeIndices] = 1
> > +      outputs["predictedActiveCells"].fill(0)
> > +      outputs["predictedActiveCells"][predictedActiveIndices] = 1
>
> So replace "predictedActiveCells" with "predictedCells"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/nupic/pull/2206/files#r33481503>.
>



-- 
*With kind regards,*

David Ray
Java Solutions Architect

*Cortical.io <http://cortical.io/>*
Sponsor of:  HTM.java <https://github.com/numenta/htm.java>

d.ray@cortical.io
http://cortical.io